sherpa-onnx/csrc/sherpa-onnx-offline.cc

```diff
@@ -4,14 +4,139 @@
 
 #include <stdio.h>
 
+#include <atomic>
 #include <chrono>  // NOLINT
+#include <fstream>
+#include <mutex>
+#include <map>
 #include <string>
+#include <thread>
 #include <vector>
 
 #include "sherpa-onnx/csrc/offline-recognizer.h"
 #include "sherpa-onnx/csrc/parse-options.h"
 #include "sherpa-onnx/csrc/wave-reader.h"
 
+std::atomic<int> wav_index(0);
+std::mutex mtx;
+
+std::vector<std::vector<std::string>> split_to_batchsize(
+    std::vector<std::string> input, int batch_size) {
+  std::vector<std::vector<std::string>> outputs;
+  auto itr = input.cbegin();
+  int process_num = 0;
+
+  while (process_num + batch_size <= static_cast<unsigned>(input.size())) {
+    auto chunk_end = itr + batch_size;
+    outputs.emplace_back(itr, chunk_end);
+    itr = chunk_end;
+    process_num += batch_size;
+  }
+  if (itr != input.cend()) {
+     outputs.emplace_back(itr, input.cend());
+  }
+  return outputs;
+}
+
+
+std::vector<std::string> load_scp_file(std::string wav_scp_path) {
+  std::vector<std::string> wav_paths;
+  std::ifstream in(wav_scp_path);
+  if (!in.is_open()) {
+      fprintf(stderr,"Failed to open file: %s.\n", wav_scp_path.c_str());
+      return wav_paths;
+  }
+  std::string line;
+  while(std::getline(in, line))
+  {
+      std::istringstream iss(line);
+      std::string column1, column2;
```

Comment:
Please move colum1 and column2 to the outside of this while loop.

Also, please use `clang-format` to reformat your code.